src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java

```diff
@@ -1472,32 +1493,15 @@ private void updatePathForExistingSolrDocs(DvObject object) throws SolrServerExc
             }
         } catch (Exception ex) {
             logger.info("failed to find dataverseSegments for dataversePaths for " + SearchFields.SUBTREE + ": " + ex);
-        }
+        }        
         List<String> dataversePaths = getDataversePathsFromSegments(dataverseSegments);
-        // Add Paths for linking dataverses
-        List <Dataverse>linkingDataverses = new ArrayList();
-        if (dataset != null){
-            linkingDataverses = dsLinkingService.findLinkingDataverses(dataset.getId());
-        } else{
-            linkingDataverses = dvLinkingService.findLinkingDataverses(dv.getId());
-        }
-        for (Dataverse linkingDataverse : linkingDataverses) {
-            List<String> linkingDataversePathSegmentsAccumulator = new ArrayList<>();
-            List<String> linkingdataverseSegments = findPathSegments(linkingDataverse, linkingDataversePathSegmentsAccumulator);
-            List<String> linkingDataversePaths = getDataversePathsFromSegments(linkingdataverseSegments);
-            for (String dvPath : linkingDataversePaths) {
-                dataversePaths.add(dvPath);
-            }
-        }
-
-        //Add paths for my linking dataverses
-        List<String> linkingDataversePaths = new ArrayList();
-        if (dataset != null) {
-            linkingDataversePaths = findLinkingDataversePaths(dataset.getOwner());          
-        } else {
-            linkingDataversePaths = findLinkingDataversePaths(dv);
-        }
-
+        /*
+        First get all linking dataverses
+        Then get their respective paths
+        And finally add to the "normal" path
+        */
+        List<Dataverse> linkingDataverses = findAllLinkingDataverses(dvo);
+        List<String> linkingDataversePaths = findLinkingDataversePaths(linkingDataverses);
         for (String dvPath : linkingDataversePaths) {
             dataversePaths.add(dvPath);
```

Comment:
can this be simplfied? since findLinkingDataversePaths reutns a List<Strings> just add that directly to dataversePaths (with addAll) instead of looping through?